services/users/src/main/java/org/opfab/users/services/CurrentUserWithPerimetersService.java

```diff
@@ -30,6 +30,20 @@ public CurrentUserWithPerimetersService(UsersService usersService, UserSettingsS
         this.entityRepository = entityRepository;
     }
 
+    public CurrentUserWithPerimeters fetchUserWithPerimeters(String login) {
+        Optional<User> userResp = usersService.fetchUserByLogin(login);
+        CurrentUserWithPerimetersData currentUserWithPerimetersData = new CurrentUserWithPerimetersData();
+
+        if (userResp.isPresent()) {
+            User user = userResp.get();
+            currentUserWithPerimetersData.setUserData(user);
+            handleUserSettings(user, currentUserWithPerimetersData);
+            handleGroups(user, currentUserWithPerimetersData);
+            handleEntities(user);
+        }
+        return currentUserWithPerimetersData;
+    }
```

Comment:
to avoid duplicated code  , get the user object and if it exists call fetchCurrentUserWithPerimeters(user) 
Add a unit test